lib/orangelight/browse_lists.rb

```diff
@@ -7,151 +7,248 @@
 
 module BrowseLists
   class << self
-    def connection
-      config = Orangelight::Application.config.database_configuration[::Rails.env]
-      dbhost = config['host']
-      dbuser = config['username']
-      dbname = config['database']
-      password = config['password']
-      sql_command = "PGPASSWORD=#{password} psql -U #{dbuser} -h #{dbhost} #{dbname} -c"
-
-      # changes for different facet queries
-      facet_request = "#{core_url}select?q=*%3A*&fl=id&wt=json&indent=true&defType=edismax&facet.sort=asc&facet.limit=-1&facet.field="
-      solr_url = Blacklight.connection_config[:url]
-      # solr_url = 'http://lib-solr1.princeton.edu:8985'
-
-      conn = Faraday.new(url: solr_url) do |faraday|
+    def connection_uri
+      Orangelight.browse_lists_index.connection.uri
+    end
+
+    def solr_client
+      Orangelight.browse_lists_index.connection
+    end
+
+    def blacklight_solr_connection
+      Blacklight.connection_config[:url]
+    end
+
+    # Isn't this the same as above?
+    def blacklight_core_url
+      Blacklight.default_index.connection.uri
+    end
+
+    def blacklight_http_client
+      @blacklight_http_client ||= Faraday.new(url: blacklight_core_url) do |faraday|
         faraday.options[:open_timeout] = 2000
         faraday.options[:timeout] = 2000
-        faraday.request  :url_encoded             # form-encode POST params
-        faraday.response :logger                  # log requests to STDOUT
-        faraday.adapter  Faraday.default_adapter  # make requests with Net::HTTP
+        faraday.request(:url_encoded)
+        faraday.response(:logger)
+        faraday.adapter(Faraday.default_adapter)
       end
-      [sql_command, facet_request, conn]
     end
```

Comment:
This should be replaced with `Blacklight.default_index.connection`, as this wraps the same client.